modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py

```diff
@@ -2393,11 +2393,30 @@ def test_dict(self):
         indices = pyarrow.array([0, 1, 0, 1, 2, 0, None, 2])
         dictionary = pyarrow.array(["first", "second", "third"])
         dict_array = pyarrow.DictionaryArray.from_arrays(indices, dictionary)
-        at = pyarrow.table({"col": dict_array})
+        at = pyarrow.table(
+            {"col1": dict_array, "col2": [1, 2, 3, 4, 5, 6, 7, 8], "col3": dict_array}
+        )
         pdf = at.to_pandas()
         mdf = from_arrow(at)
         df_equals(mdf, pdf)
 
+        mdt = mdf.dtypes[0]
+        pdt = pdf.dtypes[0]
+        assert mdt == "category"
+        assert isinstance(mdt, pandas.CategoricalDtype)
+        assert pandas.api.types.is_categorical_dtype(mdt)
+
+        if type(mdt) != pandas.CategoricalDtype:
+            # This is a lazy proxy.
+            # Make sure the table is not materialized after
+            # the instance check and comparison with string.
+            assert mdt._table is not None
+
+        assert mdt == pdt
+        assert pdt == mdt
+        assert str(mdt) == str(pdt)
```

Comment:
can we put `str` case in the "lazy" section of the test? We would like to print `df.dtypes` that trigger `str(dtype)` for every without materialization

Reply:
I'm not sure this is really required. If str() is called, most probably it's called by the user code and the user wants to see the actual values.